### PR TITLE
Fix distillation with HuBERT

### PR DESCRIPTION
**Pull request by marcoyang1998**
This PR solves https://github.com/k2-fsa/icefall/issues/779.

Due to the change of naming conventions in lhotse, the old codebook indexes can not be merged with the original training manifest. The newly generated codebook index uses lhotse version of 1.11.0. Please re-run your `prepare.sh` with the same version before proceeding with `distillation_with_hubert.sh` to avoid the potential ID mismatch problem.

### CODE REVIEW

File: egs/librispeech/ASR/distillation_with_hubert.sh [hunk 1]
```diff
@@ -155,8 +155,16 @@ if [ $stage -le 2 ] && [ $stop_stage -ge 2 ]; then
     fi
     log "Downloading extracted codebook indexes to $codebook_download_dir"
     # Make sure you have git-lfs installed (https://git-lfs.github.com)
+    # The codebook indexes are generated using lhotse 1.11.0, to avoid
+    # potential issues, we recommend you to use the same version of lhotse
+    # to run `prepare.sh`.
+    lhotse_version=$(python3 -c "import lhotse; print(lhotse.version.__version__)")
+    if [ "$lhotse_version" != "1.11.0" ]; then
```
Comment (csukuangfj) [inline egs/librispeech/ASR/distillation_with_hubert.sh:162]: Does it support `lhotse > 1.11.0`?

Comment (marcoyang1998) [reply egs/librispeech/ASR/distillation_with_hubert.sh:162]: I've only tested with `1.11.0`, newer versions could work.

**Comment by marcoyang1998** [comment]
I've removed the strict lhotse version requirement. This is achieved by adding a piece of code to resolve the ID mismatch caused by different lhotse versions.

File: egs/librispeech/ASR/distillation_with_hubert.sh [hunk 2]
```diff
@@ -156,12 +156,10 @@ if [ $stage -le 2 ] && [ $stop_stage -ge 2 ]; then
...
     # Make sure you have git-lfs installed (https://git-lfs.github.com)
     # The codebook indexes are generated using lhotse 1.11.0, to avoid
-    # potential issues, we recommend you to use the same version of lhotse
-    # to run `prepare.sh`.
-    lhotse_version=$(python3 -c "import lhotse; print(lhotse.version.__version__)")
-    if [ "$lhotse_version" != "1.11.0" ]; then
-      log "Using the wrong lhotse version(Expected: 1.11.0, but using $lhotse_version). Please install lhotse 1.11.0 before proceeding. Try pip install lhotse==1.11.0"
-      exit 1
+    # potential issues, we recommend you to use lhotse version >= 1.11.0
+    lhotse_version=$(python3 -c "import lhotse; print(lhotse.version.__version__>='1.11.0)")
```
Comment (csukuangfj) [inline egs/librispeech/ASR/distillation_with_hubert.sh:160]: Please use
https://stackoverflow.com/questions/11887762/how-do-i-compare-version-numbers-in-python
to compare versions.

File: egs/librispeech/ASR/pruned_transducer_stateless6/vq_utils.py [hunk 3]
```diff
@@ -244,10 +244,32 @@ def join_manifests(self):
...
-            cuts_vq = cuts_vq.sort_like(cuts_ori)
-            for cut_idx, (cut_vq, cut_ori) in enumerate(zip(cuts_vq, cuts_ori)):
-                assert cut_vq.id == cut_ori.id
-                cut_ori.codebook_indexes = cut_vq.codebook_indexes
+            assert len(cuts_vq) == len(cuts_ori), "Cuts should have the same length!"
+            
+            if set(cut_vq.ids) == set(cuts_ori.ids):
+                cuts_vq = cuts_vq.sort_like(cuts_ori)
+                for cut_idx, (cut_vq, cut_ori) in enumerate(zip(cuts_vq, cuts_ori)):
+                    assert cut_vq.id == cut_ori.id
```
Comment (csukuangfj) [inline egs/librispeech/ASR/pruned_transducer_stateless6/vq_utils.py:253]:
```suggestion
                    assert cut_vq.id == cut_ori.id, (cut_vq.id, cut_ori.id)
```

File: egs/librispeech/ASR/distillation_with_hubert.sh (hunk 2 above)
Comment (marcoyang1998) [reply egs/librispeech/ASR/distillation_with_hubert.sh:160]: OK

**Review by csukuangfj — APPROVED** [review]

**Pull request merged**